pex/cli/commands/lock.py

```diff
@@ -397,15 +667,51 @@ def _add_update_arguments(cls, update_parser):
                 )
             ),
         )
-        cls._add_lockfile_option(update_parser, verb="update")
-        cls._add_lock_options(update_parser)
-        cls.add_json_options(update_parser, entity="lock", include_switch=False)
-        cls._add_target_options(update_parser)
-        resolver_options_parser = cls._create_resolver_options_group(update_parser)
-        resolver_options.register_repos_options(resolver_options_parser)
-        resolver_options.register_network_options(resolver_options_parser)
-        resolver_options.register_max_jobs_option(resolver_options_parser)
-        resolver_options.register_use_pip_config(resolver_options_parser)
+
+    @classmethod
+    def _add_sync_arguments(cls, sync_parser):
+        # type: (_ActionsContainer) -> None
+        sync_parser.add_argument(
+            "--venv",
+            help=(
+                "Synchronize this venv to the contents of the lock after synchronizing the "
+                "contents of the lock to the passed requirements."
+            ),
+        )
+        sync_parser.add_argument(
+            "--venv-python",
+            help="Use this interpreter to create the `--venv` if the venv doesn't exist.",
+        )
+        sync_parser.add_argument(
+            "--pip",
```

Comment:
I don't have good intuition for when this feature will be useful. I suppose people might want some recent pip in their venv so they can mess around with it, but don't want to specify it explicitly in their requirements files because it's not a code dep in their repo? 

Either way, the behavior grid you posted in an earlier comment seems reasonable to me.

Reply:
Or, the venv may be pre-existing and they don't want to blow away the Pip already there just because it isn't in the lock.